### PR TITLE
Java sdk

### PR DESCRIPTION
**Pull request by anastygnome**
In order to allow further extensions and make the code cleaner, this PR attempts to fill some of the gaps in the current SDK proposal. 
Notably 

Compile-time warnings are fixed, and unit tests are aligned. 

I also updated the gitignore to prevent adding the target folder

Only the examples remain to be adjusted, but I will wait for a review before tackling them

### CODE REVIEW

**Comment by bencroker** [comment]
Please provide a description.

**Comment by bencroker** [comment]
Thanks. @rphumulock can you please review this, since you PR’d the original SDK?

**Comment by bencroker** [comment]
@anastygnome it looks like you’ve PR’d into the wrong directory. It should go in `sdk/java`.

File: .gitignore [hunk 1]
```diff
@@ -8,4 +8,5 @@ node_modules
 datastar-website
 *_bin
 .DS_Store
-*.qtpl.go
\ No newline at end of file
+*.qtpl.go
+code/java/*/target/
```
Comment (bencroker) [inline .gitignore]: Please fix this path.

**Comment by bencroker** [comment]
Thanks for cleaning this up! 

@rphumulock can you please review this, since it modifies your initial PR?

File: .gitignore (hunk 1 above)
Comment (anastygnome) [reply .gitignore]: done

**Comment by bencroker** [comment]
Great work, thanks!

**Pull request merged**